hack/make-rules/test.sh

```diff
@@ -157,7 +157,10 @@ junitFilenamePrefix() {
   mkdir -p "${KUBE_JUNIT_REPORT_DIR}"
   local KUBE_TEST_API_NO_SLASH="${KUBE_TEST_API//\//-}"
   # This file name isn't parsed by anything, and tee needs a shorter file name.
+  # Convert strings like junit_v1,apps-v1alpha1,authentication.k8s.io-v1beta1
+  # to junit_v1,apps.k-v1a1,authentication.k-v1b1
```

Comment:
The most important part of a comment is the why, not the what! Please tell future us why you're making this change. :)